### PR TITLE
[PSL-177] sense step3 4 5

### PR DESCRIPTION
**Pull request by bacnh8**
https://pastel.wiki/en/Architecture/OpenAPI/Sense#h-3rd-aprty-application-runs-its-own-walletnode

WalletNode
- Step5-01/03: Upload data and burn txid to 3 SNs over gRPC connection opened in Step4-01/02/03
- Step5-22/24: Accept average dd_and_fingerprints from 3 SNs

SuperNode
- Step5-01/03: Provide handler to accept Sense data over gRPC connection opened in Step4-01/02/03
- Step5-04/25: Accept, validate and process image; excange dd_and_fingerprints with 2 other SNs; select the common and sign it

Work in progress
- [x] Merge and fix conflict refactor sense service: https://github.com/pastelnetwork/gonode/pull/277
- [x] Add again step3 sequence - select 3 -  SNs from top nodes
- [x] Step 4 - Modify to create mesh network with 3 selected SNs
- [x] Step 5 - modify to send burnt txid in additional to image to SNs

### CODE REVIEW

File: common/types/types.go [hunk 1]
```diff
@@ -10,4 +10,5 @@ type MeshedSuperNode struct {
 type NftRegMetadata struct {
 	CreatorPastelID string
 	BlockHash       string
+	BurnTxID        string
```
Comment (nguyen202) [inline common/types/types.go:13]: I think should define another structure for action registration, like `ActionRegMetadata`?

File: supernode/services/senseregister/task.go [hunk 2]
```diff
@@ -506,6 +506,24 @@ func (task *Task) GetRegistrationFee(_ context.Context, ticket []byte, creatorSi
 	return task.registrationFee, err
 }
 
+// ValidateBurnTxID - will validate the pre-burnt transaction ID created by 3rd party
+func (task *Task) ValidateBurnTxID(ctx context.Context) error {
+	var err error
+
+	confirmationChn := task.waitConfirmation(ctx, task.nftRegMetadata.BurnTxID, int64(task.config.PreburntTxMinConfirmations), 15*time.Second)
```
Comment (nguyen202) [inline supernode/services/senseregister/task.go:513]: should call this in task action due to this is long run action

File: walletnode/node/grpc/register_sense.go [hunk 3]
```diff
@@ -145,26 +145,27 @@ func (service *registerSense) SendRegMetadata(ctx context.Context, regMetadata *
...
 // ProbeImage implements node.RegisterArtwork.ProbeImage()
-func (service *registerSense) ProbeImage(ctx context.Context, image *artwork.File) ([]byte, error) {
+func (service *registerSense) ProbeImage(ctx context.Context, image *artwork.File) ([]byte, bool, error) {
 	ctx = service.contextWithLogPrefix(ctx)
 	ctx = service.contextWithMDSessID(ctx)
 
 	stream, err := service.client.ProbeImage(ctx)
 	if err != nil {
-		return nil, errors.Errorf("open stream: %w", err)
+		return nil, true, errors.Errorf("open stream: %w", err)
```
Comment (nguyen202) [inline walletnode/node/grpc/register_sense.go:162]: default value should be `false`

File: walletnode/node/grpc/register_sense.go [hunk 4]
```diff
@@ -145,26 +145,27 @@ func (service *registerSense) SendRegMetadata(ctx context.Context, regMetadata *
...
 	}
 	defer stream.CloseSend()
 
 	file, err := image.Open()
 	if err != nil {
-		return nil, errors.Errorf("open file %q: %w", file.Name(), err)
+		return nil, true, errors.Errorf("open file %q: %w", file.Name(), err)
 	}
 	defer file.Close()
```
Comment (nguyen202) [inline walletnode/node/grpc/register_sense.go:170]: default value should be `false`

File: walletnode/node/grpc/register_sense.go [hunk 5]
```diff
@@ -176,19 +177,19 @@ func (service *registerSense) ProbeImage(ctx context.Context, image *artwork.Fil
 		}
 
 		req := &pb.ProbeImageRequest{
-			Payload: buffer[:n],
+			Image: buffer[:n],
 		}
 		if err := stream.Send(req); err != nil {
-			return nil, errors.Errorf("send image data: %w", err)
+			return nil, true, errors.Errorf("send image data: %w", err)
```
Comment (nguyen202) [inline walletnode/node/grpc/register_sense.go:183]: default value should be `false`

File: walletnode/node/grpc/register_sense.go (hunk 3 above)
Comment (nguyen202) [inline walletnode/node/grpc/register_sense.go:162]: I'm concerning `true` is a little bit hacky here.

File: walletnode/services/senseregister/task.go [hunk 6]
```diff
@@ -107,17 +100,14 @@ func (task *Task) run(ctx context.Context) error {
 		return task.nodes.WaitConnClose(ctx, nodesDone)
 	})
 
-	// get block height + hash
-	if err := task.getBlock(ctx); err != nil {
```
Comment (nguyen202) [inline walletnode/services/senseregister/task.go:111]: I don't think should remove this?
This is a step to prepare for registration, put it in connectToTopRankNodes() is not really meaningful

File: walletnode/node/grpc/register_sense.go (hunk 3 above)
Comment (bacnh8) [reply walletnode/node/grpc/register_sense.go:162]: in fact, this return is ignore if the error !=  nil.

File: walletnode/services/senseregister/task.go (hunk 6 above)
Comment (bacnh8) [reply walletnode/services/senseregister/task.go:111]: This function is put into connectToTopRankNodes() to follow the currennt design.

File: common/types/types.go (hunk 1 above)
Comment (bacnh8) [reply common/types/types.go:19]: done, create a new type for actionreg.

**Review by nguyen202 — APPROVED** [review]
LGTM!

**Review by a-ok123 — APPROVED** [review]

**Pull request merged**